pkg/component/apiserver/types.go

```diff
@@ -0,0 +1,119 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package apiserver
+
+import (
+	"github.com/Masterminds/semver"
+	corev1 "k8s.io/api/core/v1"
+
+	gardencorev1beta1 "github.com/gardener/gardener/pkg/apis/core/v1beta1"
+	"github.com/gardener/gardener/pkg/component"
+)
+
+// Interface contains functions for a deployer for an API server built with k8s.io/apiserver.
+type Interface interface {
+	component.DeployWaiter
+	// GetAutoscalingReplicas gets the Replicas field in the AutoscalingConfig of the Values of the deployer.
+	GetAutoscalingReplicas() *int32
+	// SetAutoscalingAPIServerResources sets the APIServerResources field in the AutoscalingConfig of the Values of the
+	// deployer.
+	SetAutoscalingAPIServerResources(corev1.ResourceRequirements)
+	// SetAutoscalingReplicas sets the Replicas field in the AutoscalingConfig of the Values of the deployer.
+	SetAutoscalingReplicas(*int32)
+	// SetETCDEncryptionConfig sets the ETCDEncryptionConfig field in the Values of the deployer.
+	SetETCDEncryptionConfig(ETCDEncryptionConfig)
+}
+
+// Values contains configuration values for the API server resources.
+type Values struct {
+	// EnabledAdmissionPlugins is the list of admission plugins that should be enabled with configuration for the API server.
+	EnabledAdmissionPlugins []AdmissionPluginConfig
+	// DisabledAdmissionPlugins is the list of admission plugins that should be disabled for the API server.
+	DisabledAdmissionPlugins []gardencorev1beta1.AdmissionPlugin
+	// Audit contains information for configuring audit settings for the API server.
+	Audit *AuditConfig
+	// Autoscaling contains information for configuring autoscaling settings for the API server.
+	Autoscaling AutoscalingConfig
+	// ETCDEncryption contains configuration for the encryption of resources in etcd.
+	ETCDEncryption ETCDEncryptionConfig
+	// FeatureGates is the set of feature gates.
+	FeatureGates map[string]bool
+	// Logging contains configuration settings for the log and access logging verbosity
+	Logging *gardencorev1beta1.KubeAPIServerLogging
+	// Requests contains configuration for the API server requests.
+	Requests *gardencorev1beta1.KubeAPIServerRequests
+	// RuntimeVersion is the Kubernetes version of the runtime cluster.
+	RuntimeVersion *semver.Version
+	// WatchCacheSizes are the configured sizes for the watch caches.
+	WatchCacheSizes *gardencorev1beta1.WatchCacheSizes
+}
+
+// AdmissionPluginConfig contains information about a specific admission plugin and its corresponding configuration.
+type AdmissionPluginConfig struct {
+	gardencorev1beta1.AdmissionPlugin
+	// Kubeconfig is an optional API server the configuration of this admission plugins. The configs for some
+	// admission plugins like `ImagePolicyWebhook` or `ValidatingAdmissionWebhook` can take a reference to a API server
```

Comment:
```suggestion
	// Kubeconfig is an optional API server connection configuration of this admission plugin. The configs for some
	// admission plugins like `ImagePolicyWebhook` or `ValidatingAdmissionWebhook` can take a reference to a API server
```
We noticed that you changed the description while copying from `kube_apiserver.go`. What was the reason for it?

Reply:
I can't remember, apparently not an intentional change. Will change it back.